### PR TITLE
atftp: add init script and config file for atftpd

### PR DESCRIPTION
**Pull request by RussellSenior**
Signed-off-by: Russell Senior <russell@personaltelco.net>

Maintainer: Daniel Danzberger / @dddaniel
Compile tested: ramips, Ubiquiti EdgeRouter X, reboot-14485-g4a46f36f99
Run tested: ramips, Ubiquiti EdgeRouter X, reboot-14485-g4a46f36f99, checked that it starts

Description: the server package, atftpd, currently lacks an init script. Added one, along with a config file. atftpd likes to run as a daemon and complained when attempting to run in foreground, so my initial efforts to implement a procd-style init script failed.

### CODE REVIEW

**Comment by neheb** [comment]
Needs a PKG_RELEASE bump.

Is it possible to add a basic test.sh file for the CI? See https://github.com/openwrt/packages/pull/13413

File: net/atftp/files/atftpd.init [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+START=95
+PIDFILE=/tmp/run/atftpd.pid
+
+start() {
+	config_load atftpd
+	config_get SRV service path "/srv/tftp"
+	config_get PORT service port 69
+
+        atftpd --pidfile $PIDFILE --user root.root --port $PORT --daemon $SRV
```
Comment (neheb) [inline net/atftp/files/atftpd.init:12]: whitespace difference?

Comment (RussellSenior) [reply net/atftp/files/atftpd.init]: Fixed PKG_RELEASE. Fixed whitespace.

I don't understand the CI testing. There are two packages generated, how is that handled?  atftp --version && atftpd --version, perhaps?

**Comment by neheb** [comment]
On Fri, Sep 18, 2020, 10:41 AM Russell Senior <notifications@github.com>
wrote:

> *@RussellSenior* commented on this pull request.
> ------------------------------
>
> In net/atftp/files/atftpd.init
> <https://github.com/openwrt/packages/pull/13417#discussion_r491098218>:
>
> > @@ -0,0 +1,17 @@
> +#!/bin/sh /etc/rc.common
> +# Copyright (C) 2020 OpenWrt.org
> +
> +START=95
> +PIDFILE=/tmp/run/atftpd.pid
> +
> +start() {
> +	config_load atftpd
> +	config_get SRV service path "/srv/tftp"
> +	config_get PORT service port 69
> +
> +        atftpd --pidfile $PIDFILE --user root.root --port $PORT --daemon $SRV
>
> Fixed PKG_RELEASE. Fixed whitespace.
>
> I don't understand the CI testing. There are two packages generated, how
> is that handled? atftp --version && atftpd --version, perhaps?
>
Sure.

**Comment by aparcar** [comment]
Regarding the test script you will have the two variables "$1" containing the package name (so aftp or aftpd) and "$2" which contains the current version of the tool. You could use something like this:

```shell
"$1" --version | grep "$2"
```

You could also create a switch statement doing more sophisticated tests.

**Comment by RussellSenior** [comment]
Added a runtime test according to the suggestion of @aparcar.

**Comment by neheb** [comment]
```
Use package specific test.sh
atftp-0.7.2 (client)
Test failed
```

**Comment by aparcar** [comment]
@neheb not bad, 12 seconds faster than me!

**Comment by aparcar** [comment]
The problem is likely that aftp sends it to `stderr`. So `"$1" 2>&1 | grep "$2"` should fix it.

**Comment by neheb** [comment]
Strange that they only fail on arm and x86...

**Comment by aparcar** [comment]
Runtest is only supported on a few architectures, I only implemented it for x86 and arm. We can add it to more, ping me in chat for that.

**Comment by RussellSenior** [comment]
Since you have control over the CI testing infrastructure, and I just want to be able to use the damn server package, and it clearly works, maybe one of you CI enthusiasts can fix the test.sh script.

**Comment by aparcar** [comment]
>fix the test.sh script

But you just fixed it yourself! The issue was only `stderr`. The testing of additional architectures doesn't block this PR, no worries.

> CI enthusiasts

Always on duty! Thanks for the quick fixups @RussellSenior

**Comment by aparcar** [comment]
```shell
Line 12:
        atftpd --pidfile $PIDFILE --user root.root --port $PORT --daemon $SRV
                                                          ^-- SC2086: Double quote to prevent globbing and word splitting.
>>                                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
        atftpd --pidfile $PIDFILE --user root.root --port "$PORT" --daemon "$SRV"
 
Line 16:
        kill $(cat $PIDFILE)
             ^-- SC2046: Quote this to prevent word splitting.
```

Do you mind fixing these shellcheck complains? I'm happy to merge it afterwards

**Pull request merged**